core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -94,6 +95,12 @@ public function __construct(RequestStack $requestStack, ScopeMatcher $scopeMatch
         $this->errorLevel = $errorLevel;
     }
 
+    public function reset(): void
+    {
+        $this->adapterCache = [];
+        $this->isFrontend = false;
```

Comment:
What about `$this->hookListeners`? And do we not have to reset `self::$initialized` as well?

Reply:
`self::$initialized` explicitly is a static so it's persistent across instances. I guess we can't change that because of the constants.

`$this->hookListeners` are not supposed to be added multiple times, they don't have state, they just initialize the service. Re-doing that wouldn't change anything.